runtime/rasdump/dmpsup.c

```diff
@@ -1365,7 +1366,45 @@ initSystemInfo(J9JavaVM *vm)
 			}
 		}
 	}
-	appendSystemInfoFromFile(vm, J9RAS_SYSTEMINFO_CORE_PATTERN, J9RAS_CORE_PATTERN_FILE);
+	{
+		J9RASSystemInfo *corePatternInfo = appendSystemInfoFromFile(vm, J9RAS_SYSTEMINFO_CORE_PATTERN, J9RAS_CORE_PATTERN_FILE);
+		if (NULL != corePatternInfo) {
+			/* A common core_pattern is Dynatrace; for example, |/opt/dynatrace/oneagent/agent/rdp
+			 * This program sends the core to the originally configured
+			 * core_pattern as stored in, for example,
+			 * /opt/dynatrace/oneagent/agent/conf/original_core_pattern
+			 *
+			 * If we find this Dynatrace core_pattern, extract its installation
+			 * directory and then read original_core_pattern relative to that.
+			 */
+			const char *corePattern = (const char *)corePatternInfo->data;
+			if ('|' == corePattern[0]) {
+				static const char search[] = "/oneagent/agent/rdp";
+				static const char replacement[] = "/oneagent/agent/conf/original_core_pattern";
+				const char *dynatracePath = strstr(corePattern, search);
```

Comment:
As I read this again, I realize the length test below may be too restrictive, unless the pattern should be expected to _end_ with `/oneagent/agent/rdp`. Should the use of `strstr()` be replaced by an equivalent of `String.endsWith()`?

Reply:
> unless the pattern should be expected to _end_ with `/oneagent/agent/rdp`.

No, it's not expected to end with `/oneagent/agent/rdp`. It will almost always have some tokens ([for example](https://www.dynatrace.com/support/help/platform-modules/applications-and-microservices/profiling-and-optimization/crash-analysis#operating-system-changes), `/opt/dynatrace/oneagent/agent/rdp -p %p -e %e -s %s`).

If we have `sizeof(namebuf) -  1` (one for `\0`) available, the prefix length is `strlen(corePattern) - strlen(dynatracePath) - 1`  (one for `|`), and we'll need `sizeof(replacement)` appended to the prefix, then re-arranging would lead to:

```
sizeof(namebuf) > strlen(corePattern) + strlen(dynatracePath) + sizeof(replacement)
```

Does that seem right?

Reply:
Sorry, I meant:

```
sizeof(namebuf) > strlen(corePattern) - strlen(dynatracePath) + sizeof(replacement)
```

Reply:
After rearranging suitably, I think this is what we want:
```c
				/* Check if core_pattern includes the Dynatrace agent. */
				if (NULL != dynatracePath) {
					char namebuf[MAX_INTERESTING_LENGTH];
					/* The length of the agent path prefix, minus the pipe character. */
					size_t prefixLength = dynatracePath - corePattern - 1;

					/* Ensure that the original core_pattern path will fit in our buffer. */
					if (prefixLength <= (sizeof(namebuf) - sizeof(replacement))) {
						/* Copy starting after the pipe character. */
						memcpy(namebuf, corePattern + 1, prefixLength);

						/* Append the location of original_core_pattern. */
						memcpy(namebuf + prefixLength, replacement, sizeof(replacement));

						/* Finally, read the file contents. */
						appendSystemInfoFromFile(vm, J9RAS_SYSTEMINFO_CORE_ORIGINAL_PATTERN, namebuf);
					}
				}
```
Note the second use of `memcpy()` (instead of `strcpy()`) and that there are no calls to `strlen()` now (we don't care about the lengths of `corePattern` or `dynatracePath`).

Reply:
Ah yes, that makes sense. Thank you.